src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -331,6 +330,31 @@ private static bool IsStaticClass(Type type)
             return type.GetTypeInfo().IsAbstract && type.GetTypeInfo().IsSealed;
         }
 
+        private object cancelLock = new object();
+
+        /// <summary>
+        /// Cancel (abort or stop) a CompositeWorkItem and all of its children
+        /// </summary>
+        /// <param name="force">true if the CompositeWorkItem and all of its children should be aborted, false if it should allow all currently running tests to complete</param>
+        public override void Cancel(bool force)
+        {
+            lock (cancelLock)
+            {
+                if (_children == null)
+                    return;
+
+                foreach (var child in _children)
+                {
+                    var ctx = child.Context;
+                    if (ctx != null)
+                        ctx.ExecutionStatus = force ? TestExecutionStatus.AbortRequested : TestExecutionStatus.StopRequested;
+
+                    if (child.State == WorkItemState.Running)
+                        child.Cancel(force);
+                }
+            }
+        }
```

Comment:
Am I missing something here? It looks like you do nothing for the current work item context but only set the child context execution status to Abort/RunRequested. Why do it for the current item context?